source/obs/gs/gs-effect.cpp

```diff
@@ -27,11 +27,11 @@
 
 #define MAX_EFFECT_SIZE 32 * 1024 * 1024 // 32 MiB, big enough for everything.
 
-static std::string load_file_as_code(std::filesystem::path shader_file, bool is_top_level = true)
+static std::string load_file_as_code(const std::filesystem::path& shader_file, bool is_top_level = true)
 {
 	std::stringstream     shader_stream;
-	std::filesystem::path shader_path = std::filesystem::absolute(shader_file);
-	std::filesystem::path shader_root = std::filesystem::path(shader_path).remove_filename();
+	const std::filesystem::path& shader_path = std::filesystem::absolute(shader_file);
+	const std::filesystem::path& shader_root = std::filesystem::path(shader_path).remove_filename();
```

Comment:
I'm unsure if this does what you intend it to do. This clones shader_path temporarily, and removes the filename from it. It then would have to be std::move()d into the local variable, but instead this now creates a reference to memory that is considered free. VS 2022 even flagged this as a use-after-free bug.